src/get-trip-id-stop-ids-lookup.js

```diff
@@ -0,0 +1,50 @@
+const { createReadStream } = require('node:fs');
+const { resolve } = require("path");
+const { parse } = require('csv-parse');
+
+
+/**
+ * Loops over `stop_times.txt` and generates a lookup table, where the key is a trip-id,
+ * and the value is an Array of stop-ids on that trip.
+ * 
+ * @param {*} unzippedGtfsPath 
+ * @returns {Object} {
+ *    tripIdStopIdsLookup: <trip-id>: <stop-id>[],
+ *    stopIdTripIdsLookup: Map<<stop-id>, Set<trip-id>>
+ * }
+ */
+async function getStopsForTripLookup(unzippedGtfsPath){
+  const stopTimesStream = createReadStream(resolve(unzippedGtfsPath, 'stop_times.txt'), {encoding: 'utf8'});
+  const parser = stopTimesStream.pipe(parse({columns: true}));
+  const tripIdStopIdsLookup = {};
+  // Don't need to serialize the <stop-id> : <trip-ids>[] lookup, so use a Map
+  const stopIdTripIdsLookup = new Map();
```

Comment:
You could just as well have them both be maps here, since you iterate over the object-as-map and change it below.

Reply:
Yea, but again I don't wanna allocate a full copy of this dictionary, just to mutate one of its props.|

This talks is the root of most of my philosophy around this
http://youtube.com/watch?v=rX0ItVEVjHc&t=975s

TLDR: Good memory management makes things go fast!

Reply:
Is the map big enough this matters?